core/src/mender-api.c

```diff
@@ -446,6 +457,11 @@ mender_api_publish_deployment_status(const char *id, mender_deployment_status_t
     char        *response     = NULL;
     int          status       = 0;
 
+    if (MENDER_OK != (ret = mender_api_ensure_authenticated())) {
```

Comment:
What if it returns `MENDER_DONE` (i.e., already authenticated)? __NB:__ _There are also some other places I saw this._

Reply:
Good catch! I wonder if I should change the function to just return `MENDER_OK` in all successful cases. Or maybe we could have a macro checking for errors vs. good `mender_err_t` values?

Reply:
> I wonder if I should change the function to just return MENDER_OK in all successful cases.

Yepp, can't think of any immediate reason for why the caller should really care about whether it already was authenticated or not